ansible_base/resource_registry/models/resource.py

```diff
@@ -40,6 +41,21 @@ def can_be_managed(self):
     def get_resource_config(self):
         return self.resource_registry.get_config_for_model(model=ContentType.objects.get_for_id(self.content_type_id).model_class())
 
+    def get_conflicting_resource(self, resource_data):
+        filter = {}
+
+        serialized_data = self.serializer_class(data=resource_data)
+        serialized_data.is_valid(raise_exception=True)
+        serialized_data = serialized_data.validated_data
+
+        for field in self.serializer_class.UNIQUE_FIELDS:
+            filter[field] = serialized_data[field]
+
+        try:
+            return Resource.get_resource_for_object(self.content_type.get_object_for_this_type(**filter))
```

Comment:
Any possibility that we could get a different type of exception here? Or if we do raising is ok?

Reply:
Potentially "too many objects returned", but if that's the case, then the model is misconfigured and I'd rather raise it.